metron-platform/metron-parsers/src/main/java/org/apache/metron/parsers/GrokParser.java

```diff
@@ -150,8 +149,8 @@ public void init() {
 
       if (message.size() == 0)
         throw new RuntimeException("Grok statement produced a null message. Original message was: "
-                + originalMessage + " and the parsed message was: " + message + " . Check the pattern at: "
-                + grokPath);
+                + originalMessage + " , parsed message was: " + message + " , pattern was: "
+                + grokPattern);
```

Comment:
You have a point here.  The whole pattern shouldn't be logged on every failure, especially since a bad grok pattern will get logged for every message.  However there is a reason I did it that way.  As a developer, I prefer to have everything I need to troubleshoot something in the same log line.  If we log a pattern only when it changes it will be hard to tell which grokPattern caused a failure.  You have the message but you'll have to go digging in the logs to find the grokPattern that was applied.  Can you think of a way to log efficiently but still provide the context needed to troubleshoot an issue?